pkg/symbolizer/symbolizer.go

```diff
@@ -319,55 +389,243 @@ func (s *Symbolizer) Symbolize(ctx context.Context, locations []*pb.Location) er
 
 // symbolizeLocationsForMapping fetches the debug info for a given build ID and symbolizes it the
 // given location.
-func (s *Symbolizer) symbolizeLocationsForMapping(ctx context.Context, m *pb.Mapping, locations []*pb.Location) ([][]profile.LocationLine, error) {
-	logger := log.With(s.logger, "buildid", m.BuildId)
-
-	// Fetch the debug info for the build ID.
-	rc, err := s.debuginfo.FetchDebuginfo(ctx, m.BuildId)
+func (s *Symbolizer) symbolizeLocationsForMapping(ctx context.Context, m *pb.Mapping, locations []*pb.Location) ([][]profile.LocationLine, liner, error) {
```

Comment:
Is it possible to split this method into two? One that returns a liner and another one consumes that an returns location lines?